### PR TITLE
improved the if statement

### PR DESCRIPTION
**Pull request by shivam-Purohit**
## PR Checklist

- [x] Read the [contribution guidelines](/CONTRIBUTING.md)
- [ ] Added a brief description of these changes to the top of the [changelog](/CHANGELOG.md)
- [x] Changes in code meet test criteria (running `mvn test` returns exit code 0, without errors)

## PR Type

- [x] Refactoring (changes that do not affect API nor functionality)

## Summary
Improved the if statement to make the code shorter and easy to understand

### CODE REVIEW

**Review by NyorJa — CHANGES_REQUESTED** [review]
I suggest you use apache commons lang 3 and StringUtils.equals

It has built in null checking and comparison

**Comment by shivam-Purohit** [comment]
should I need to add the dependencies.
I added ` <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-lang3</artifactId>
      <version>3.12.0</version>
    </dependency>
` in pom.xml
and this is my if statement code
`   if (!StringUtils.isEmpty(phone1) && !StringUtils.equals(target.getphone1(),phone1)) {
        target.setPhone1(phone1);
      }`
is this what you Okay?

**Comment by NyorJa** [comment]
> 
`!StringUtils.equals(target.getphone1(),phone1)` is ok. It has null checking on it.

image

**Comment by shivam-Purohit** [comment]
@NyorJa should I remove the null checking part then .
` if (!StringUtils.equals(target.getphone1(),phone1)) { target.setPhone1(phone1); } `
This should do I guess then.

**Comment by NyorJa** [comment]
> @NyorJa should I remove the null checking part then . `if (!StringUtils.equals(target.getphone1(),phone1)) { target.setPhone1(phone1); }` This should do I guess then.

Yes. That is correct.

**Comment by shivam-Purohit** [comment]
You can have a look and suggest if anymore changes required.

File: src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java [hunk 1]
```diff
@@ -72,11 +73,11 @@ public Person applyChangesToExistingEntity(PersonPojo source, Person existing) t
 
     // phones may be empty, but not null
     String phone1 = source.getPhone1();
-    if (phone1 != null) {
-      if (!target.getPhone1().equals(phone1)) {
+
+      if ( !StringUtils.equals(target.getphone1(),phone1)) {
```
Comment (NyorJa) [inline src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java:77]: Please remove the extra space :)

Comment (NyorJa) [reply src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java:77]: in line 77

it should be 

`if (!StringUtils.equals(target.getphone1(),phone1))`

**Comment by NyorJa** [comment]
Hi @mepox @bglamadrid, 

I think this is good. Please confirm. Thanks.

**Review by mepox — CHANGES_REQUESTED** [review]
Hi @shivam-Purohit ! Please see my comment.

File: src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java [hunk 2]
```diff
@@ -72,11 +73,11 @@ public Person applyChangesToExistingEntity(PersonPojo source, Person existing) t
 
     // phones may be empty, but not null
     String phone1 = source.getPhone1();
-    if (phone1 != null) {
-      if (!target.getPhone1().equals(phone1)) {
+
+      if (!StringUtils.equals(target.getphone1(),phone1)) {
         target.setPhone1(phone1);
       }
```
[the remark this answers is not in the document]
Comment (NyorJa) [reply src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java:79]: Thanks @mepox for double checking. Are you using an IDE @shivam-Purohit ? What I mean is you are not experiencing a compilation error?

[the remark this answers is not in the document]
Comment (shivam-Purohit) [reply src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java:79]: `     String phone1 = source.getPhone1();
    if (!StringUtils.equals(target.getPhone1(),phone1)) {
        target.setPhone1(phone1);
      } `

Screenshot (93)

I was using IntelliJ but not showing any errors Idk why?
checkout the code and sorry for the repeated errors.

I also removed the extra line at 79

[the remark this answers is not in the document]
Comment (NyorJa) [reply src/main/java/org/trebol/jpa/services/conversion/PeopleConverterJpaServiceImpl.java]: Oww..I think we just saw here it in this thread. We should check under files changed. Your changes are good now.

@bglamadrid please check

**Pull request merged**